### PR TITLE
detect some legacy Windows versions and return legacy update version

### PR DESCRIPTION
**Pull request by mgallien**
should allow to serve old desktop client versions to legacy operating
systems

### CODE REVIEW

File: config/config.php [hunk 1]
```diff
@@ -33,8 +35,10 @@
 	$mac_url = $url . 'Mac/Installer/';
 } else {
 	$url = 'https://github.com/nextcloud/desktop/releases/download/v' . $ver . '/';
+	$legacyurl = 'https://github.com/nextcloud/desktop/releases/download/v' . $legacy_ver . '/';
```
Comment (nickvergessen) [inline config/config.php:38]: Iirc versions before 3.1.x cant follow redirects and therefor need to be served from download.nextcloud.com

Comment (er-vin) [reply config/config.php:38]: Correct. This won't work for those.

Comment (mgallien) [reply config/config.php:38]: Thanks both, I will fix that

**Comment by mgallien** [comment]
If I understand correctly I will need to update Brander to generate oem config files compatible with this change

File: src/Response.php [hunk 2]
```diff
@@ -89,12 +101,41 @@ private function getUpdateVersion() : array {
 			$this->updateSegment = random_int(0, 99);
 		}
 
+		$useNewerVersion = true;
+		if ($this->platform === 'win32') {
+			if ($this->osVersion === '10') {
+				switch ($this->kernelVersion) {
```
Comment (allexzander) [inline src/Response.php:107]: @mgallien This looks complicated. Shouldn't we just split `10.0.xxxxx` into 3 components, and then, just check if the last part `xxxxx` is greater or equal to 1709?

**Comment by mgallien** [comment]
no longer needed

**Pull request closed without merging**